deepset_cloud_sdk/_service/files_service.py

```diff
@@ -299,6 +303,101 @@ async def upload(
             show_progress=show_progress,
         )
 
+    async def _download_and_log_errors(
+        self,
+        workspace_name: str,
+        file_id: UUID,
+        file_name: str,
+        file_dir: Optional[Union[Path, str]],
+        include_meta: bool,
+    ) -> None:
+        try:
+            await self._files.download(
+                workspace_name=workspace_name,
+                file_id=file_id,
+                file_name=file_name,
+                file_dir=file_dir,
+                include_meta=include_meta,
+            )
+        except FileNotFoundInDeepsetCloudException as e:
+            logger.error("File was listed in deepset Cloud but could not be downloaded.", file_id=file_id, error=e)
+        except Exception as e:
+            logger.error("Failed to download file.", file_id=file_id, error=e)
+
+    async def download(
+        self,
+        workspace_name: str,
+        file_dir: Optional[Union[Path, str]] = None,
+        name: Optional[str] = None,
+        content: Optional[str] = None,
+        odata_filter: Optional[str] = None,
+        include_meta: bool = True,
+        batch_size: int = 50,
+        timeout_s: Optional[int] = None,
+        show_progress: bool = True,
+    ) -> None:
+        """Download a folder to deepset Cloud.
+
+        :param workspace_name: Name of the workspace to upload the files to. It uses the workspace from the .ENV file by default.
+        :param file_dir: Path to the folder to download. If None, the current working directory is used.
+        :param name: odata_filter by file name.
+        :param content: odata_filter by file content.
+        :param odata_filter: odata_filter by file meta data.
+        :param include_meta: If True, downloads the metadata files as well.
+        :param batch_size: Batch size for the listing.
+        :param limit: Limit for the listing.
+        :param show_progress: Shows the upload progress.
+        """
+        start = time.time()
+        logger.info("Start downloading files.", workspace_name=workspace_name)
+
+        pbar: Optional[tqdm] = None
```

Comment:
optional: You could use the `NullPattern` to handle the optional tdqm nicer https://blog.devops.dev/design-patterns-in-python-null-object-pattern-82e79617a141

Reply:
Makes sense. Lets use it for the next ones! Thanks for sharing!